higher-order-functions/min-max-sorted.md

```diff
@@ -0,0 +1,342 @@
+#    "release_year": 1999
+#}
+```
+
+We can use the same pattern to find the movie with the earliest release date and the first title alphabetically.
+
+```py
+def get_release_year(movie):
+    return movie["release_year"]
+
+def get_title(movie):
+    return movie["title"]
+
+earliest_movie = min(movies, key=get_release_year)
+# earliest_movie = {
+#    "title": "Recursion",
+#    "genre": "Intrigue",
+#    "rating": 2.0,
+#    "release_year": 1999
+#}
+
+first_movie_by_title = min(movies, key=get_title)
+
+# first_movie_by_title = {
+#    "title": "2 JavaScript 2 React",
+#    "genre": "Action",
+#    "rating": 4.2,
+#    "release_year": 2015
+#}
+```
+
+## Keyword Arguments
+
+### !callout-info
+
+## Keyword Arguments
+
+The `key` parameter is an example of a keyword argument. 
+
+### !end-callout
+
+Thus far we've defined functions with positional arguments. The table below describes the difference between defining and calling functions with positional and keyword arguments. 
+
+| Type | Function Definition | Function Call |
+| -- | --| --|
+| Positional Arguments | `def function_name(parameter1, parameter2)` | `function_name(argument1, argument2)`|
+| Keyword Arguments |`def function_name(parameter1, parameter2)` | `function_name(parameter1=argument1, parameter2=argument2)` |
+
+Functions can be called with a mix of keyword and positional arguments, but positional arguments cannot follow keyword arguments in the function call. To learn more, [here is an article we recommend](https://treyhunner.com/2018/04/keyword-arguments-in-python/).
+
+Consider the following function:
+
+```py
+def sum(a, b):
+    return a + b
+```
+
+The following table describes valid and invalid syntax for calling `get_the_sum`:
+
+| Function Call | Valid Syntax (yes/no) | Explanation |
+|--|--|--|
+|`sum(1, 2)` | yes | This is an example of calling a function with two positional arguments. |
+|`sum(a=1, b=2)` | yes | This is an example of calling a function with two keyword arguments. |
+|`sum(1, b=2)` | yes | This is an example of calling a function with a mix of positional and keyword arguments. |
+|`sum(a=1, 2)` | no | This is invalid. Positional arguments cannot follow keyword arguments. <br> `SyntaxError: positional argument follows keyword argument`|
+
+## Higher Order Functions
+
+The `min` function with the `key` parameter is an example of a higher order function.
+
+### !callout-info
+
+## Higher Order Functions
+
+A higher order function is a function that takes another function as an argument.
+
+### !end-callout
+
+A function is an object in memory that is a set of instructions. A function name is a variable that points to this set of instructions. 
+
+When we use the `()` syntax, we are calling the address of that variable. When we leave off the `()`, we are looking up the address that the variable is pointing to.
+
+Just like variables that point to other objects (strings, integers, lists, etc.), we can assign a function to another variable:
```

Comment:
I don't think it's _necessary_ for understanding, but we could add a little clarity about where `get_title` is coming from in the example below 

```suggestion
Just like variables that point to other objects (strings, integers, lists, etc.), we can assign a function to another variable. Assuming `get_title` is a function as defined in the example above, we could write:
```